pkg/provider/azure_loadbalancer.go

```diff
@@ -919,11 +919,18 @@ func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service,
 
 func (az *Cloud) findMatchedPIPByLoadBalancerIP(service *v1.Service, loadBalancerIP, pipResourceGroup string, pips *[]network.PublicIPAddress) (*network.PublicIPAddress, error) {
 	if pips == nil {
+		return nil, fmt.Errorf("findMatchedPIPByLoadBalancerIP: nil pip list passed")
+	} else if cap(*pips) == 0 {
+		// use capacity to check whether pip has been listed
```

Comment:
Is it an implicit requirement when we call this function? It seems to be error-prone.

Reply:
why not using `len()` here? (`len()` is used in other places)

Reply:
do you think it makes sense to save the listed PIP in an in-memory cache instead of passing the pointer to other functions? e.g. something like vmss cache or vm cache?

Reply:
updated, thanks